main.go

```diff
@@ -160,3 +166,30 @@ func backwardsCompatibleMetricFactory(configKey string) func(xmetrics.MetricsIn)
 		return registry, nil
 	}
 }
+
+type OnErrorHTTPResponseOption struct {
+	AuthType string
+}
```

Comment:
Usually structs are at the top of a file.  It may also make sense to make a new `errorHTTPResponse.go` file?